Sources/SwiftGRPCNIO/GRPCClientConnection.swift

```diff
@@ -49,8 +50,23 @@ open class GRPCClientConnection {
       }
 
     return bootstrap.connect(host: host, port: port).flatMap { channel in
-      channel.pipeline.context(handlerType: HTTP2StreamMultiplexer.self).map { context in
-        context.handler as! HTTP2StreamMultiplexer
+      // Check the handshake succeeded and a valid protocol was negotiated via ALPN.
+      let tlsVerified: EventLoopFuture<Void>
+
+      if case .none = tlsMode {
+        tlsVerified = channel.eventLoop.makeSucceededFuture(())
+      } else {
+        tlsVerified = channel.pipeline.context(handlerType: GRPCTLSVerificationHandler.self).map {
+          $0.handler as! GRPCTLSVerificationHandler
```

Comment:
I've seen this pattern in the previous "hang fix" PR as well. Should this functionality maybe be provided by NIO itself? /cc @weissi

Reply:
There's already a PR open for this but it will require NIO 2.1: https://github.com/apple/swift-nio/pull/974

Reply:
Awesome; could you make a note to yourself to update our code once that becomes available?

Reply:
Added TODOs.